src/mpid/ch4/generic/am/mpidig_am_recv.c

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (C) by Argonne National Laboratory
+ *     See COPYRIGHT in top-level directory
+ */
+
+#include "mpidimpl.h"
+#include "mpidig_am_recv.h"
+
+void MPIDIG_deferred_am_irecv_dequeue(MPIDIG_deferred_am_irecv_req_t * dreq)
+{
+    MPIDIG_deferred_am_irecv_req_t *curr_dreq = dreq;
+
+    MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDIG_DEFERRED_AM_IRECV_DEQUEUE);
+    MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDIG_DEFERRED_AM_IRECV_DEQUEUE);
+
+    DL_DELETE(MPIDI_global.deferred_am_irecv_q, curr_dreq);
+    MPL_free(curr_dreq);
+
+    MPIR_FUNC_VERBOSE_EXIT(MPID_STATE_MPIDIG_DEFERRED_AM_IRECV_DEQUEUE);
+}
+
+int MPIDIG_deferred_am_irecv_issue(MPIDIG_deferred_am_irecv_req_t * dreq, int *status)
+{
+    int mpi_errno = MPI_SUCCESS;
+    MPIR_Comm *root_comm;
+
+    MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDIG_DEFERRED_AM_IRECV_ISSUE);
+    MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDIG_DEFERRED_AM_IRECV_ISSUE);
+
+    root_comm = MPIDIG_context_id_to_comm(dreq->context_id);
+    mpi_errno = MPIDIG_prepare_recv_req(dreq->rank, dreq->tag, dreq->context_id, dreq->buf,
+                                        dreq->count, dreq->datatype, dreq->rreq);
+    if (mpi_errno != MPI_SUCCESS) {
+        *status = MPIDIG_AM_IRECV_DEFERRED;
+        goto fn_exit;
+    }
+
+    MPIR_Comm_add_ref(root_comm);
+    MPIDIG_enqueue_posted(dreq->rreq, &MPIDIG_COMM(root_comm, posted_list));
+    *status = MPI_SUCCESS;
```

Comment:
Why is status using `MPI_SUCCESS`?  It's should use an enum that's consistent with `MPIDIG_AM_IRECV_DEFERRED`.  Also, it uses a hack that assumes that `MPIDIG_AM_IRECV_DEFERRED` is always non-zero, so it doesn't conflict with `MPI_SUCCESS`.